### PR TITLE
Kill processes after completing build and before publishing test results or artifacts

### PR DESCRIPTION
**Pull request by dougbu**
- aspnet/AspNetCore-Internal#1735
- port aspnet/Extensions@e2bb8ac1a0 fix into this repo
  - change `ci-official.yml` and `default-build.yml` instead of `azure-pipelines.yml`

### CODE REVIEW

**Comment by natemcmaster** [comment]
We should do this in ci.yml and delete ci-official.yml from 2.x branches. It's not used and I removed that file from the master branch in favor of ci.yml.

**Comment by dougbu** [comment]
> We should do this in ci.yml

`ci.yml` doesn't even contain `buildScript` settings at the moment. Alright to remove `ci-official.yml` and stick with my changes to the `default-build.yml` template?

**Comment by ryanbrandenburg** [comment]
The OOM error on linux seems unrelated. @natemcmaster is that tracked somewhere? I seem to remember you mentioning something similar.

**Review by ryanbrandenburg — COMMENTED** [review]
The code seems fine. I'll be happy when @natemcmaster is happy.

**Comment by dougbu** [comment]
> The OOM error on linux seems unrelated.

Agreed, I copied from the wrong tab. The bug related to this PR is #4067.

**Comment by natemcmaster** [comment]
> remove ci-official.yml and stick with my changes to the default-build.yml template?

Yes sounds good to me

**Review by natemcmaster — APPROVED** [review]

File: eng/scripts/KillProcesses.ps1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+taskkill /T /F /IM dotnet.exe
+taskkill /T /F /IM testhost.exe
+taskkill /T /F /IM iisexpress.exe
+taskkill /T /F /IM iisexpresstray.exe
+taskkill /T /F /IM w3wp.exe
+taskkill /T /F /IM msbuild.exe
+taskkill /T /F /IM vbcscompiler.exe
+taskkill /T /F /IM git.exe
+taskkill /T /F /IM vctip.exe
+taskkill /T /F /IM chrome.exe
```
Comment (natemcmaster) [inline eng/scripts/KillProcesses.ps1]: Can you add `h2spec.exe`? We've had issues with this on occasion.

File: eng/scripts/KillProcesses.ps1 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env pwsh -c
```
[the remark this answers is not in the document]
Comment (dougbu) [reply eng/scripts/KillProcesses.ps1]: :ok: you've convinced me 😺 I'll remove this and do `chmod -x` on the file.

File: eng/scripts/KillProcesses.ps1 (hunk 1 above)
Comment (dougbu) [reply eng/scripts/KillProcesses.ps1]: Sure

**Review by dougbu — COMMENTED** [review]
Once this goes through, I'll run through the aspnet/Extensions branches to make `KillProcesses.ps1` consistent.

**Pull request merged**